ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/LoiCardAdapter.kt

```diff
@@ -44,63 +45,58 @@ class LoiCardAdapter : RecyclerView.Adapter<ViewHolder>() {
 
   /** Binds [LocationOfInterest] data to [ViewHolder]. */
   override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-    holder.bind(itemsList[position])
+    val loi: LocationOfInterest = itemsList[position]
+
+    holder.bind(loi)
 
     // Add highlight border if selected.
     holder.binding.wrapperView.background =
       ResourcesCompat.getDrawable(
         holder.itemView.context.resources,
-        if (selectedIndex == position) {
+        if (focusedIndex == position) {
           R.drawable.border
         } else {
           R.color.colorBackground
         },
         null
       )
 
-    // Handle action buttons.
-    holder.binding.loiCard.setOnClickListener { handleItemClicked(position) }
-    holder.binding.start.setOnClickListener { handleButtonClicked(position) }
-    holder.binding.review.setOnClickListener { TODO() }
-    holder.binding.markComplete.setOnClickListener { TODO() }
+    val isCaptionMissing = loi.caption.isNullOrEmpty()
+    holder.binding.loiIcon.visibility = if (isCaptionMissing) View.GONE else View.VISIBLE
+    holder.binding.loiName.visibility = if (isCaptionMissing) View.GONE else View.VISIBLE
+    holder.binding.loiCard.setOnClickListener { collectDataCallback.invoke(loi) }
   }
 
   /** Returns the size of the list. */
   override fun getItemCount() = itemsList.size
 
-  fun updateSelectedPosition(newSelectedIndex: Int) {
-    if (newSelectedIndex < 0 || newSelectedIndex >= itemCount || selectedIndex == newSelectedIndex)
-      return
-    handleItemClicked(newSelectedIndex)
+  /** Updates the currently focused item. */
+  fun focusItemAtIndex(newIndex: Int) {
+    if (newIndex < 0 || newIndex >= itemCount || focusedIndex == newIndex) return
+
+    focusedIndex = newIndex
+    notifyDataSetChanged()
+
+    cardFocusedCallback.invoke(itemsList[newIndex])
   }
 
+  /** Overwrites existing cards. */
   fun updateData(newItemsList: List<LocationOfInterest>) {
     itemsList.clear()
     itemsList.addAll(newItemsList)
-    selectedIndex = -1
+    focusedIndex = -1
     notifyDataSetChanged()
-    cardSelectedCallback.invoke(null)
+    cardFocusedCallback.invoke(null)
   }
 
-  fun setLoiCardSelectedCallback(callback: (LocationOfInterest?) -> Unit) {
-    this.cardSelectedCallback = callback
+  fun setLoiCardFocusedCallback(callback: (LocationOfInterest?) -> Unit) {
+    this.cardFocusedCallback = callback
```

Comment:
Should we call this `*Listener` for consistency with Android SDK callbacks?

Reply:
Done